src/backends/github/API.js

```diff
@@ -343,10 +365,18 @@ export default class API {
 
   deleteUnpublishedEntry(collection, slug) {
     const contentKey = slug;
-    let prNumber; 
     return this.retrieveMetadata(contentKey)
     .then(metadata => this.closePR(metadata.pr, metadata.objects))
-    .then(() => this.deleteBranch(`cms/${ contentKey }`));
+    .then(() => this.deleteBranch(`cms/${ contentKey }`))
+    // If the PR doesn't exist, then this has already been deleted -
+    // deletion should be idempotent, so we can consider this a
+    // success.
+    .catch((err) => {
+      if (err.message === "Reference does not exist") {
```

Comment:
The error code here is a 422, and there's no other metadata available there other than the error message. [422 is a very general error code in the GitHub API](https://developer.github.com/v3/#client-errors), so resolving this promise for all 422s will other errors, so I checked the more specific error message.

https://developer.github.com/v3/#client-errors says clients should differentiate 422 errors based on the body of the response (the `APIError` wrapper we use isn't including this information, which should probably be changed in another PR). I was hoping there would be more metadata in the request, but when inspecting a relevant 422 response manually, only the `message` and `documentation_url` were present, leaving us with just the `message` and status to switch between. Given GitHub's API documentation, it seems that the `message` is intended to be used to differentiate errors, so I used that.

If you'd rather catch all 422 errors, I can change this pretty trivially (just need to change the condition to `err.status === 422`).